src/openrct2/ride/TrackDesignRepository.cpp

```diff
@@ -319,7 +320,7 @@ class TrackDesignRepository final : public ITrackDesignRepository
             {
                 return a.RideType < b.RideType;
             }
-            return String::Compare(a.Name, b.Name) < 0;
+            return strlogicalcmp(a.Name.c_str(), b.Name.c_str()) < 0;
```

Comment:
We should have a C++ equivalent for this.

Reply:
I agree, std::string does have an actual compare function

Reply:
But using std:: string compare would revert the "logical" sorting, wouldn't it? (Not arguing against am std::string overload though)

Reply:
I've never heard about a "logical" string sort, what makes it special?

Reply:
I wonder if what this is supposed to achive could be:
```
    std::sort(strings.begin(), strings.end(),[](auto& a, auto& b)->bool
    {
        return !std::lexicographical_compare(a.begin(), a.end(),
            b.begin(), b.end(),
            [](auto& c1, auto& c2)
            {
                return std::tolower(c1) < std::tolower(c2);
            });
    });
```

Reply:
@ZehMatt It should be immediately obvious if you take a look at the `strlogicalcmp` function: it sorts `Mechanic 5` before `Mechanic 10` rather than after it, like regular sort would.

Reply:
See my comment above.

Reply:
I'm all for your solution if it achieves the same effect, I just reused what we already had

Reply:
I'm also wondering, is the sort actually working correctly with other languages than lets say en-gb/en-us/de-de etc. Like Korean/Russian/Chinese etc. Another aspect is the UTF8 encoding, using tolower would be the wrong thing here. I never really looked at what it does until now and I'm starting to think that the implementation is incorrect. Also regarding the logical sort, the common term is lexicographical, and we should change that at some point.

Reply:
@ZehMatt I integrated ICU for character set conversions￼, but also for Unicode-aware case transformations. We should probably use it more throughout the code base.